bundles/framework/hierarchical-layerlist/view/LayerGroupTab.js

```diff
@@ -980,6 +985,14 @@ Oskari.clazz.define(
             };
 
             me.tabPanel.getContainer().append(layerTree);
+
+            if(jQuery('.hierarchical-layerlist-search-noresults').length === 0) {
```

Comment:
This searches from the whole document. Please change it to find under the flyout/tabpanel.